### DISCUSSION TITLE
Prisma Crashes with nested GraphQL query that contains a one-to-many relationship

### DISCUSSION THREAD

**Discussion by MEnnabah** [prisma/prisma#18800] [Q&A]
Hi everyone

We’re using GraphQL, and we have the following query:

```gql
type Cat {
  id: ID!
  name: String!
  age: Int!
  
  legs: [Leg!]!
}

type Leg {
  id: ID!
  color: String!
  catId: Int!
  claws: [Claw!]!
}

type Claw {
  id: ID!
  length: String!
  legId: Int!
}

type Query {
  cat(id: Int!): Cat!
}
```

Prisma crashes when using findFirst, findUnique or $transaction for the path cat.legs.claws because legs is in an array. If `cat.legs` wasn’t an array (as opposed to `leg`), it won’t crash.

I have already opened an [issue](https://github.com/prisma/prisma/issues/18787), but I’m looking if anyone has encountered this before, and what did you do.

Thanks!

**Comment by nurul3101** [comment]
Hello @MEnnabah 👋 

Thank you for taking the time to create an issue. Our engineering team monitors those issues and can get back on the issue.

Do you get this error if you Query for only the `cat` and `legs` without the claws?

**Comment by MEnnabah** [comment]
Hi @nurul3101

Thank you for taking the time to take a look at this!

> Do you get this error if you Query for only the `cat` and `legs` without the `claws`?

No. The crash only happens when querying nested arrays. Because `cat` is not an array, querying `legs` only is ok.

However, if the query has `legs`, and it has `claws`, it will crash. Because `legs` is an array, and `claws` is an array.

**Reply by nurul3101** [reply]
For reference: https://github.com/prisma/prisma/issues/18787#issuecomment-1515152817

Continuing the conversation in the issue that you created.
